packages/react-core/src/components/TimePicker/TimePicker.tsx

```diff
@@ -366,6 +366,9 @@ export class TimePicker extends React.Component<TimePickerProps, TimePickerState
       this.onToggle(true);
     }
     e.stopPropagation();
+    this.setState({
+      isInvalid: false
+    });
```

Comment:
This might fall into the category of nitpick/personal preference, but I'm wondering whether we should only set the `isInvalid` state on blur period rather than also setting it to false on focus?

Mainly not sure whether having the error message/icon appear and disappear all the time would be visually confusing; it looks like the message/icon also appears when going from focus on the input to navigating into the menu options with a keyboard.

Reply:
I did so because I want to make sure when we select the option from the dropdown menu, the error message could disappear. Maybe it's better to set `isInvalid: false` when selecting the option from the dropdown?

Reply:
Ah gotcha, that makes sense! Setting `isInvalid` to false when selecting from the dropdown I'm thinking could work to remove the error when a user selects what should be a valid option.